ledger-core/virtual/integration/method_test.go

```diff
@@ -343,6 +344,90 @@ func TestVirtual_Method_WithoutExecutor_Unordered(t *testing.T) {
 	mc.Finish()
 }
 
+func TestVirtual_Method_WithoutExecutor_Ordered(t *testing.T) {
+	t.Log("C5093")
+
+	mc := minimock.NewController(t)
+
+	server, ctx := utils.NewUninitializedServer(nil, t)
+	defer server.Stop()
+
+	runnerMock := logicless.NewServiceMock(ctx, t, nil)
+	server.ReplaceRunner(runnerMock)
+
+	server.Init(ctx)
+	server.IncrementPulseAndWaitIdle(ctx)
+
+	var (
+		class        = testwallet.GetClass()
+		objectLocal  = server.RandomLocalWithPulse()
+		objectGlobal = reference.NewSelf(objectLocal)
+	)
+
+	Method_PrepareObject(ctx, server, payload.Ready, objectGlobal)
+	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+
+	cntr := 0
+	countBefore := server.PublisherMock.GetCount()
+
+	{
+		typedChecker.VCallResult.Set(func(res *payload.VCallResult) bool {
+			require.Equal(t, res.ReturnArguments, []byte("345"))
+			require.Equal(t, res.Callee, objectGlobal)
+			return false // no resend msg
+		})
+		interferenceFlag := contract.CallTolerable
+		stateFlag := contract.CallDirty
+
+		for i := int64(0); i < 2; i++ {
+			callOutgoing := server.BuildRandomOutgoingWithPulse()
+			pl := payload.VCallRequest{
+				CallType:            payload.CTMethod,
+				CallFlags:           payload.BuildCallFlags(interferenceFlag, stateFlag),
+				Caller:              server.GlobalCaller(),
+				Callee:              objectGlobal,
+				CallSiteDeclaration: class,
+				CallSiteMethod:      "ordered" + strconv.FormatInt(i, 10),
+				CallOutgoing:        callOutgoing,
+			}
+
+			result := requestresult.New([]byte("345"), objectGlobal)
+
+			key := callOutgoing.String()
+			runnerMock.AddExecutionMock(key).
+				AddStart(func(ctx execution.Context) {
+					cntr ++
+					for k := 0; k < 5; k ++ {
+						require.Equal(t, 1, cntr)
+						time.Sleep(3 * time.Millisecond)
+					}
+					cntr --
+			}, &execution.Update{
+				Type:   execution.Done,
+				Result: result,
+			})
+			runnerMock.AddExecutionClassify(key, contract.MethodIsolation{
+				Interference: interferenceFlag,
+				State:        stateFlag,
+			}, nil)
+
+			server.SendPayload(ctx, &pl)
+		}
+	}
+	awaitFullStop := server.Journal.WaitStopOf(&execute.SMExecute{}, 2)
```

Comment:
please move that part to place before messages were sent (as we discussed before)

Reply:
Right. Done.